plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt

```diff
@@ -219,19 +221,15 @@ fun AwsBearerTokenConnection.lazyIsUnauthedBearerConnection(): Boolean {
 
 fun reauthConnectionIfNeeded(project: Project?, connection: ToolkitConnection): BearerTokenProvider {
     val tokenProvider = (connection.getConnectionSettings() as TokenConnectionSettings).tokenProvider.delegate as BearerTokenProvider
-    val needReAuth = maybeReauthProviderIfNeeded(project, tokenProvider) {
-        ApplicationManager.getApplication().executeOnPooledThread {
```

Comment:
with current implementation, by using `executeOnPoolThread`, we have no way to run logic onAuthSuccess or onAuthFailure. Therefore revert this change.

Reply:
the original intention of this change is to hide the progress indicator while users are going through the browser auth flow